.github/workflows/upgrade-testing.yml

```diff
@@ -0,0 +1,167 @@
+name: Upgrade Tests
+
+on:
+  push:
+    branches:
+      - trunk
+    # Always test the workflow after it's updated.
+    paths:
+      - '.github/workflows/upgrade-testing.yml'
+      - '.github/workflows/upgrade-testing-run.yml'
+  pull_request:
+    # Always test the workflow when changes are suggested.
+    paths:
+      - '.github/workflows/upgrade-testing.yml'
+      - '.github/workflows/upgrade-testing-run.yml'
+  workflow_dispatch:
+    inputs:
+      new-version:
+        description: 'The version to test updating to.'
+        type: string
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name for pull requests
+  # or the commit hash for any other events.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
+jobs:
+  #
+  # Spawns upgrade testing for WordPress 6.x versions with MySQL.
+  #
+  upgrade-tests-wp-6x-mysql:
+    name: ${{ matrix.wp }} to ${{ inputs.new-version && inputs.new-version || 'latest' }}
+    uses: WordPress/wordpress-upgrade-tester/.github/workflows/upgrade-testing-run.yml@combine-upgrade-test-workflows
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    permissions:
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest' ]
+        php: [ '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2' ]
+        db-type: [ 'mysql' ]
+        db-version: [ '5.7', '8.0' ]
+        wp: [ '6.0', '6.1', '6.2' ]
```

Comment:
Since 6.x supports 7.0-8.2, we may need to split this one into a new job at 6.5 to keep the number low enough.

Reply:
Yeah we'll likely need to keep a close eye on the PHP minimum version support. Although this may change again in the coming releases as PHP 7.0 and 7.1 support continues to drop. We may find ourselves in a "sweet spot" right now, but your point stands as a consideration for the near/far future